### ISSUE TITLE
Repeater lockup

### ISSUE THREAD

**Issue by Marvin-YYC** [tuxphone/CubeCellRepeater#8]
Thanks again for creating this code and sharing it with the world.

I have been seeing my repeater lockup or freeze, sometimes a couple times a day. During this freeze there is no activity from the RXTX led.   I have also noticed that the green RX led occasionally stays on steady bright for several minutes at a time.  I am not sure if this is followed by a device freeze.   I tried monitoring these issues by connecting it by serial,  but these issues don't seem to occur while connected by serial.  

Other than setting my region US (North America), I have not altered any of the code.

Any suggestions would be appreciated.

**Comment by tuxphone** [comment]
Hello Marvin, thanks for reporting.

Could you give some information about your "environment"? Do you have a lot of active nodes in your area or is it rather quiet?
Can you rule out issues with your battery? 

It could be a problem with waking up after entering sleep mode, but i do have a repeater node whose run time is only limited by the battery (no solar panel). 

Did you turn off serial output?

To turn off serial output #define SILENT at the beginning of main.h:

```
// No Crypto = all zero
#define SILENT
```

**Comment by Marvin-YYC** [comment]
My other nodes are reporting contact with about 53 nodes in my area. There seems to be a lot of traffic.  

The green or green+red led usually blinks a few times each minute during the day. (As I mentioned sometimes the green stays on bright until another RX cycles it off. I think that's what's happening)

The battery is good, in fact I'm getting more than 48 hours out of a single 2600mAh 18650. Just put in a new battery this morning. No solar yet.  

I'll try turning off the serial output and see if that helps. . . I'll let you know.  

Many Thanks.  :)

**Comment by Marvin-YYC** [comment]
Well;  I'm not too good at this stuff, but when I uncommented the _#define SILENT_ in _main.h_ I end up with two errors in _main.cpp_.  

When I comment  _#define SILENT_ again, I end up with just the first error.
Don't know what I am doing wrong?? I checked it against the codein github and it is the same??

	ERRORS Shown below
	
	"owner": "cpp",
	"severity": 8,
	"message": "expected primary-expression before ';' token",
	"startLineNumber": 207,
	"startColumn": 101,
	"endLineNumber": 207,
	"endColumn": 101

	"owner": "C/C++: IntelliSense",
	"code": "29",
	"severity": 8,
	"message": "expected an expression",
	"source": "C/C++",
	"startLineNumber": 207,
	"startColumn": 101,
	"endLineNumber": 207,
	"endColumn": 102

**Comment by tuxphone** [comment]
It looks like you accidentally deleted something.
Perhaps download the package again and try to compile with uncommented #define SILENT at the top of main.h.

**Comment by Marvin-YYC** [comment]
Well; I downloaded the package again as suggested, but same problems persists.  

Each time I un-comment the #define SILENT at the top of main.h I get the same two errors.  Only difference now is that when I re-comment the #define SILENT both errors disappear this time.   

I have been able to flash it to my CubeCell, but still having the same problem with the green led staying on bright, and the lock up.

**Comment by tuxphone** [comment]
It compiles for me. No problems at all, with region code US or EU868. SILENT or not. GCC 9.21 (standard cubecell), gcc 12.3.1 (default setting in platform.ini) and gcc 13.2.1.

Are you running it on a Linux machine? (I'm on Windows)

**Comment by Marvin-YYC** [comment]
Interesting.   I'm running Windows.  

Not sure what could be wrong at my end. 

I've been thinking of managing it with an ESP32,  so I can monitor the battery and case temperature. (I'll be able to connect to WiFi)   If i can figure that out, perhaps I can figure out a way for the esp32 to restart the repeater once in a while.  

I also discovered that some bad actor was sending out neighbour information every 15 seconds in my area.  Might have been part of the problem with the locking up.

**Comment by tuxphone** [comment]
Perhaps you could attach your main.h and main.cpp so i can inspect/run on my machine.

Should not differ, but still worth a shot.

**Comment by Marvin-YYC** [comment]
OK Thanks

The repeater was still locking up every couple of hours. So I moved to a location that had a poor line of sight. It has now been running for over 24 hours without freezing up.  I'm going to try reducing the power setting and see if that helps in reducing the traffic while still using a good LOS location.  

I am also getting this when I run it, but I'm not sure how to update  the 'pip'.  Perhaps this is part of the trouble?!??
_[notice] A new release of pip is available: 24.1.1 -> 24.2
[notice] To update, run: python.exe -m pip install --upgrade pip_

[maincpp.zip](https://github.com/user-attachments/files/16416897/maincpp.zip)
[mainh.zip](https://github.com/user-attachments/files/16416898/mainh.zip)

**Comment by tuxphone** [comment]
I have just published a new branch without serial output (led only).

Change the branch from "master" to "RepeatAll" and download the code if you want to give it a try,

I have identified and removed 2 possible reasons for lockups, too.

**Comment by Marvin-YYC** [comment]
Great.  I'll give it a try.  :)

**Comment by Marvin-YYC** [comment]
Unfortunately I had to reset the repeater this morning as it had locked up over night. It had been running since about noon and had locked up sometime after 5:00am. I also noticed that even though you have reduced the TX/RX LED output, the green LED frequently turns on very brightly and stays on for several seconds sometimes for more than 60 seconds, as it was doing prior to the changes. 

Update: 
 It froze up again about six hours after the last reset :(
Another freeze up again over night, after several hours, this time the green LED stayed on very bright.  I did not reset it right away. The green LED stayed very bright for 5 hours.   Connected to serial in this state and there was no activity.

**Comment by tuxphone** [comment]
There is a possible lockup, when trying to set RX and it fails. I mitigated it with today's commit.

Now, when activating RX fails it resets the SX1262 and tries again. It may be possible to interrupt a running TX with this change, but i guess 1 skipped sending is better than a frozen unit.

Please try it and report your findings.

**Comment by Marvin-YYC** [comment]
I had trouble getting it to compile and haven't had time lately to take a closer look at what I'm doing wrong.  I'll hopefully have some time soon.  Thanks again for your efforts and the update.

**Comment by Marvin-YYC** [comment]
I finally got back to this.  Was able to get it to compile no problem this time.
I am still seeing the bright green LED some times stay on for a minute or two, but I am not seeing it lock-up anymore.  It's been running for a few hours with lots of traffic.

To try to figure out what is going on with the green light, I changed the status lights -  Rx to Red and Tx to Blue, but still seeing the green come on at full brightness.  Red and blue are at 16 and fairly dim.   

Also; I tried to enable SILENT mode, but get an error every time  (see below)
When I comment out the SILENT it clears the error.

>  expected primary-expression before ';' token  [Ln 186, Col 74]
>  expected an expression  [Ln 186, Col 74]
> 

 >     line 186  (err == RADIOLIB_ERR_NONE) ? MSG("success!\n\r") : MSG("failed!\n\r");

**Comment by Marvin-YYC** [comment]
Update about the green LED . . . .
Now that I have a couple more Cubecells, I flashed two more boards and see that the green light does not come on at all.  Must have a defective board.  Also;  now I see that the power status light sometimes flashes bright when connected to usb power.  Will have to try some more testing with batteries.  
Thanks

**Comment by Marvin-YYC** [comment]
So I switched to another board, and now it has the bright green LED flash too. Possibly due to being connected to a battery?!?! This board also locks up occasionally. . .  Frustrating :/

I did notice one thing; your code is written for the CubeCell HTCC AB01 - my boards are CubeCell HTCC AB01-V2.  Does this make a difference?  I haven't figured out how to change the config to the V2 yet.  I'm new at PIO
